packages/astro/src/astro-components/control/SignedOutCSR.astro

```diff
@@ -0,0 +1,25 @@
+<clerk-signed-out hidden>
+  <slot />
+</clerk-signed-out>
+
+<script>
+import { type AuthState, BaseClerkControlElement } from './BaseClerkControlElement';
+
+class ClerkSignedOut extends BaseClerkControlElement {
+  protected onAuthStateChange(state: AuthState): void {
+    if (state.userId) {
+      this.setAttribute('hidden', '');
+    } else {
+      this.removeAttribute('hidden');
+    }
+  }
+}
+
+customElements.define('clerk-signed-out', ClerkSignedOut);
+</script>
+
+<style>
+  clerk-signed-out[hidden] {
+    display: none !important;
```

Comment:
In case someone would like to animate in/out the children of `<SignedOut/>` would this be possible ?

Reply:
Updated the control components to have `class` props so they can override it. We'll mention this in the CSR reference docs when added 👍🏼